app/lib/features/chat/providers/notifiers/chat_typing_notifier.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:async';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/settings/providers/app_settings_provider.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::room::typing_notice');
+
+class ChatTypingEventStateNotifier extends StateNotifier<TypingEvent?> {
+  final Ref ref;
+  ChatTypingEventStateNotifier(this.ref) : super(null) {
+    _init();
+  }
+  StreamSubscription<TypingEvent?>? _poller;
+  void _init() async {
+    final userAppSetttings = await ref.read(userAppSettingsProvider.future);
+    final enabled = userAppSetttings.typingNotice();
+    final client = ref.read(alwaysClientProvider);
+    _poller = client.typingEventRx()!.listen((e) {
+      state = e;
+    });
+    if (enabled == null) {
+      // default settings not available, enable it automatically.
+      final updater = userAppSetttings.updateBuilder();
+      updater.typingNotice(true);
+      await updater.send();
+      configure(true);
+    } else {
+      configure(enabled);
+    }
```

Comment:
we should _not_ overwrite the settings, we should just consider 'null' to be doing what we do by default. In particular these type of side-effects should not be happening when you initialize a notifier!
```suggestion
```

Reply:
So that means, initially these settings should not have any set value OR you're proposing this isn't a great time/area to have it set up?

Reply:
we shouldn't set this _unless_ the user interacts with it. and unitl then interpret the `null` as `true`.